### ISSUE TITLE
There was a problem executing this request

### ISSUE THREAD

**Issue by jmurphy45** [chrisbarr/AmazonSNS-PHP-API#3]
I have a progrma in the format of this and I am getting this error There was a problem executing this request. Could you halp me understand the problem.

	// Create an instance
	$AmazonSNS = new AmazonSNS('key', 'key');
	$AmazonSNS->setRegion('us-east-1');
	
	// And send a message to subscribers of this topic
	$AmazonSNS->publish(topicarn', $message);

**Comment by chrisbarr** [comment]
Hi, you need to create a topic before you are able to publish messages to it, so something like the following:

```php
$AmazonSNS = new AmazonSNS('key', 'key');
$topicArn = $AmazonSNS->createTopic('my_topic');

$AmazonSNS->publish($topicArn, $message);
```

Have a look at the [Readme](https://github.com/chrisbarr/AmazonSNS-PHP-API/blob/master/README.md) and [Amazon's Docs](http://docs.aws.amazon.com/sns/latest/api/Welcome.html)

**Comment by jmurphy45** [comment]
Well I already have a topic created.

**Comment by chrisbarr** [comment]
Ok, it's a little hard to tell without a complete example ;)
I suggest adding `var_dump($info, $output);` to line 468 in AmazonSNS.php and try running it again - Amazon should (hopefully) return a useful error message.

**Comment by ctg3** [comment]
I had the same issue. I put in the var_dump and then grabbed the http request. I put that in the browser and discovered that the user I was using didn't have the proper policy in IAM.

**Comment by ctg3** [comment]
Actually, that didn't solve my issue. I did have an issue with IAM, but that didn't resolve my "There was a problem executing this request" error. I was getting a result code 0 from curl. So I did var_dump(curl_error($ch)); From this I discovered I had an curl SSL issue. I downloaded the CA cert from here: http://curl.haxx.se/docs/caextract.html and then set curl_setopt($ch, CURLOPT_CAINFO, 'cacert.pem');
